user-registration-app/src/app/models/registration.ts

```diff
@@ -174,4 +174,38 @@ export class RegistrationApiResponse {
   isCurrentRegistration(): boolean {
     return this.hackathon.active;
   }
+  parseFromApiResponse(value: RegistrationApiResponse) : Registration{
+    const registration = new Registration();
+    registration.firstName = value.firstname;
+    registration.lastName = value.lastname;
+    registration.gender = value.gender;
+    registration.email = value.email;
+    registration.eighteenBeforeEvent = !!value.eighteenBeforeEvent;
+    registration.shirtSize = value.shirt_size;
+    registration.dietaryRestriction = value.dietary_restriction;
+    registration.allergies = value.allergies;
+    registration.travelReimbursement = !!value.travel_reimbursement;
+    registration.firstHackathon = !!value.first_hackathon;
+    registration.university = value.university;
+    registration.phone = value.phone;
+    registration.academicYear = value.academic_year;
+    registration.major = value.major;
+    registration.resume = value.resume;
+    registration.mlhcoc = !!value.mlh_coc;
+    registration.mlhdcp = !!value.mlh_dcp;
+    registration.ethnicity = value.race;
+    registration.codingExperience = value.coding_experience;
+    registration.referral = value.referral;
+    registration.projectDesc = value.project;
+    registration.expectations = value.expectations;
+    registration.veteran = value.veteran;
+    registration.submitted = !!value.submitted;
+    registration.hackathon = this.parseHackathon(registration);
+    registration.pin = (this.pinAi-value.pin).toString();
```

Comment:
@sushshring Once I move this up into Registration, this.pinAi doesn't exist anymore, I've been assuming that value.pin is the pin of the hackathon and not the pin of the user which is what registration.pin is meant to be. Is that right?